holoviews/operation/downsample.py

```diff
@@ -176,7 +181,9 @@ def _process(self, element, key=None):
             return element.clone(elements)
 
         if self.p.x_range:
-            element = element[slice(*self.p.x_range)]
+            xs = element.dimension_values(0)
+            x0, x1 = (np.argmin(np.abs(xs-coord)) for coord in self.p.x_range)
+            element = element.iloc[max(x0 - 1, 0):(x1 + 2)]
```

Comment:
What was wrong with the previous approach?

Reply:
This seems like bad practice because it means you are forcing out-of-core data to be loaded into memory just to apply this slicing. If the previous approach didn't work properly then the data interfaces need to be fixed instead of implementing a custom solution here.

Reply:
> What was wrong with the previous approach?

This could leave white space at the start/end of the plot (using the original example). Indicates that there is no more data, even when there is.  

![image](https://github.com/holoviz/holoviews/assets/19758978/1f682868-493a-460c-9dab-a59beac8a408)

> This seems like bad practice because it means you are forcing out-of-core data to be loaded into memory just to apply this slicing.

Noted, but this is still very much WIP, so it is not ready for a critical review. Even more, this is pretty harshly written, which could have been better framed as a question.  

>  If the previous approach didn't work properly then the data interfaces need to be fixed instead of implementing a custom solution here.

I don't think there is anything wrong with the data interface, here I just want the next datapoints outside of the viewport. This could be upstreamed to the data interface, but I'm not sure about the implementation yet.

Reply:
Sounds good, I think we have long needed but never implemented snapping semantics for the `DataInterface.select` operation. Implementing it for all data interfaces is definitely a good bit of work though.

>Noted, but this is still very much WIP, so it is not ready for a critical review. Even more, this is pretty harshly written, which could have been better framed as a question.

That certainly wasn't the intent, I think without having that consideration being explicitly pointed out there's much chance that anyone would have considered this without having dask and ibis backends top-of-mind. So it was meant as a heads up, not a criticism.